models/stics/R/write.config.STICS.R

```diff
@@ -0,0 +1,395 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the 
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+##-------------------------------------------------------------------------------------------------#
+##' Writes STICS configurations.
+##'
+##' Requires a pft xml object, a list of trait values for a single model run,
+##' and the name of the file to create
+##'
+##' @name write.config.STICS
+##' @title Write STICS configuration files
+##' @param defaults list of defaults to process
+##' @param trait.values vector of samples for a given trait
+##' @param settings list of settings from pecan settings file
+##' @param run.id id of run
+##' @importFrom dplyr "%>%"
+##' @return configuration file for STICS for given run
+##' @export
+##' @author Istem Fer
+##-------------------------------------------------------------------------------------------------#
+write.config.STICS <- function(defaults, trait.values, settings, run.id) {
+  
+  ## simulation days, used later
+  dseq <- seq(lubridate::as_date(settings$run$start.date), lubridate::as_date(settings$run$end.date), by = "day")
+  
+  # find out where to write run/ouput
+  rundir  <- file.path(settings$host$rundir, run.id)
+  pltdir  <- file.path(settings$host$rundir, run.id, "plant")
+  cfgdir  <- file.path(settings$host$rundir, run.id, "config")
+  bindir  <- file.path(settings$host$rundir, run.id, "bin")
+  outdir  <- file.path(settings$host$outdir, run.id)
+  
+  
+  ## create plant, config and bin dirs
+  dir.create(pltdir)
```

Comment:
Is rundir guaranteed to exist already? If not, want recursive = TRUE here too

Reply:
they are usually guaranteed by write.configs functions upstream e.g. https://github.com/PecanProject/pecan/blob/develop/base/utils/R/sensitivity.R#L193 and https://github.com/PecanProject/pecan/blob/develop/base/utils/R/ensemble.R#L318 but maybe for the sake of making model packages a little more independent I can check